### DISCUSSION TITLE
extension task fail to build

### DISCUSSION THREAD

**Discussion by ItaliaNorway** [AntlerVC/firetable#1436] [Support / Q&A] [answered]
I´m tring to add a extension task but I always receive error in  deploying the function to firebase...
I also tried a "hello world" funcion and I setill receive the same error.
What I do wrong? 

`=== Deploying to 'fansii'...

i  deploying functions
i  functions: preparing codebase default for deployment
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
✔  functions: required API cloudbuild.googleapis.com is enabled
✔  artifactregistry: required API artifactregistry.googleapis.com is enabled
i  functions: preparing . directory for uploading...
i  functions: packaged /home/node/app/build/functionBuilder/builds/1697484655502 (74.48 KB) for uploading
✔  functions: . folder uploaded successfully
i  functions: creating Node.js 18 function R-UserRatigs(europe-west1)...
Build failed: Build error details not available.Please check the logs at https://console.cloud.google.com/cloud-build/builds;region=europe-west1/390ae6b0-6722-4d86-96ee-a0706fed6106?project=139120552284. Please visit https://cloud.google.com/functions/docs/troubleshooting#build for in-depth troubleshooting documentation for build related errors.

Functions deploy had errors with the following functions:
	R-UserRatigs(europe-west1)
i  functions: cleaning up build files...

Error: There was an error deploying functions
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

39
21:32:42:42  

command: cd build/functionBuilder/builds/1697484655502; yarn deploy --project fansii --only functions

stderr: error Command failed with exit code 2.

errorStackTrace: Error: Command failed: cd build/functionBuilder/builds/1697484655502; yarn deploy --project fansii --only functions
error Command failed with exit code 2.

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)`

**Comment by bofeiw** [comment]
Usually the cause to the error can be found in your rowy-backend Cloud Run logs panel in GCP console (https://console.cloud.google.com/run). Do you see anything relevant?

**Comment by ItaliaNorway** [comment]
Unfortunately there are no entries in the building log…  (https://console.cloud.google.com/cloud-build/builds;region=europe-west1/5816e868-a7ba-4204-8c29-0cb6f5045653?project=fansii) 


In the https://console.cloud.google.com/run
rowy-backend <https://console.cloud.google.com/run/detail/europe-north1/rowy-backend?project=fansii>	0  	europe-north1
rowy-hooks <https://console.cloud.google.com/run/detail/europe-north1/rowy-hooks?project=fansii>	0  	europe-north1

The the function is europe-west1 
My firebase is in Europe-west1

Can be this the cause?

**Reply by bofeiw** [reply]
The function location should be fine as long as you see some build logs in Rowy app.

Can you check the logs in the logs tab here?
https://console.cloud.google.com/run/detail/europe-north1/rowy-backend?project=fansii
<img width="263" alt="Screenshot 2023-10-17 at 20 11 43" src="https://github.com/rowyio/rowy/assets/34177142/42720c64-c6ee-4f1e-b086-5cc5b49208f1">

**Comment by ItaliaNorway** [comment]
￼

**Reply by ItaliaNorway** [reply]
![image](https://github.com/rowyio/rowy/assets/35761457/26bf5f01-72ba-46e4-b1f9-3403251edccf)

**Reply by bofeiw** [reply]
This error message usually means the table function failed to deploy, which could be caused by many reasons. For example, a code typo, type error, or using variable that does not exist. Can you check the logs and see if there are anything that tells the deploy error details?

**Comment by ItaliaNorway** [comment]
I tried with a basic “hello World” code and does the same. 
I also reset the project and started from scratch. 
I’m out of options…

**Comment by Chadha93** [comment]
Hi @ItaliaNorway, let's help you resolve this issue.  

As requested by @bofeiw [here](https://github.com/rowyio/rowy/discussions/1436#discussioncomment-7319868), could you please check your logs and see if there is anything that tells the deploy error details? This will help us to debug your issue. 

Cheers,
Gaurav

**Comment by ItaliaNorway** [comment]
I tested out a completely new project with Rowy.
Blank new Firebase project started from rowy. 
Created a table with Rating (numbers) and AverageRatings (derivative) 
Listener fields: Rating
Output filed type: Number

Derivative script:
// Import any NPM package needed
// import _ from "lodash";

const derivative: Derivative = async ({ row, ref, db, storage, auth, logging }) => {
  logging.log("derivative started");
return row.rating;
  // Example:
  // const sum = row.a + row.b;
  // return sum;
};

export default derivative;

Evaluate all it works.
When I Deploy. Same error as before (in the last project): 
=== Deploying to 'luup-iuzwma'...

i  deploying functions
i  functions: preparing codebase default for deployment
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
✔  artifactregistry: required API artifactregistry.googleapis.com is enabled
✔  functions: required API cloudbuild.googleapis.com is enabled
i  functions: preparing . directory for uploading...
i  functions: packaged /home/node/app/build/functionBuilder/builds/1697723849775 (72.44 KB) for uploading
✔  functions: . folder uploaded successfully
i  functions: updating Node.js 18 function R-Functions(europe-west1)...
Build failed: failed to Fetch: failed to download archive gs://gcf-sources-1091868825787-europe-west1/R-Functions-6ffdea61-8621-4450-8a8c-5c8dd1637fb8/version-2/function-source.zip: Access to bucket gcf-sources-1091868825787-europe-west1 denied. You must grant Storage Object Viewer permission to ***@***.***


Functions deploy had errors with the following functions:
	R-Functions(europe-west1)
i  functions: cleaning up build files...

Error: There was an error deploying functions
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
38
15:58:30:30  

command: cd build/functionBuilder/builds/1697723849775; yarn deploy --project luup-iuzwma --only functions

stderr: error Command failed with exit code 2.


errorStackTrace: Error: Command failed: cd build/functionBuilder/builds/1697723849775; yarn deploy --project luup-iuzwma --only functions
error Command failed with exit code 2.

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Reply by bofeiw** [reply]
Looking at the error message, your service account has a missing permission:

> Build failed: failed to Fetch: failed to download archive gs://gcf-sources-1091868825787-europe-west1/R-Functions-6ffdea61-8621-4450-8a8c-5c8dd1637fb8/version-2/function-source.zip: Access to bucket gcf-sources-1091868825787-europe-west1 denied. You must grant Storage Object Viewer permission to ***@***.***

Can you do as the error message suggested: **grant Storage Object Viewer permission to ***@***.*****

And also check any other missing permissions for Rowy service account as documented here:https://docs.rowy.io/rowy-run#manual-install-rowy-run, assuming that you have a manual install of Rowy.

**Comment by ItaliaNorway** [comment]
I did the automatic install of Roy…
I gave “storage object view” permission now but  I remember did it also for the previous project… stil same error as the previous project called Fansii:

Build failed: Build error details not available.Please check the logs at https://console.cloud.google.com/cloud-build/builds;region=europe-west1/9e44e0fc-3837-406e-9859-49574f6c4179?project=1091868825787. Please visit https://cloud.google.com/functions/docs/troubleshooting#build for in-depth troubleshooting documentation for build related errors.

Functions deploy had errors with the following functions:
	R-Functions(europe-west1)
i  functions: cleaning up build files...

Error: There was an error deploying functions
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
38
20:20:10:10  

command: cd build/functionBuilder/builds/1697825916381; yarn deploy --project luup-iuzwma --only functions

stderr: error Command failed with exit code 2.


errorStackTrace: Error: Command failed: cd build/functionBuilder/builds/1697825916381; yarn deploy --project luup-iuzwma --only functions
error Command failed with exit code 2.

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)

**Reply by ItaliaNorway** [reply]
I tried the AI demo expense Parser .. follow the instructions. Deploy the functions at the end.
Same error:
errorStackTrace: Error: Command failed: cd build/functionBuilder/builds/1698080979934; yarn deploy --project luup-iuzwma --only functions
error Command failed with exit code 2.

I´m quite desperate. What I need to do? 
It looks like an error with any functions also the one I try from firebase CLI or the Cloud console...

**Comment by ItaliaNorway** [comment, marked as answer]
I had to start a complete new project without using ROWY. Create and deploy a funcion and then add Rowy
